CalculatorEngine.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Linq;
+
+namespace GettingStartedWithCSharp
+{
+    public class CalculatorEngine : ICalculatorEngine
+    {
+        private decimal _value = 0m;
+        private decimal _memory;
+        public bool HasMemoryStored { get; private set; } = false;
+        private readonly IMessageBoxDisplayService _messageBoxDisplayService;
```

Comment:
Nu ti-a sunat _niciun clopotel de alarma_ ca o clasa de logica de business pura trebuie sa interactioneze cu utilizatorul? This is _so bad_.

Nu e oare de datoria Presenter-ului sa intrebe utilizatorul daca e sigur ca vrea sa stearga memoria si in functie de rezultat sa puna engine-ul la munca?

Acum ai multat _prea mult_ din presenter in engine, si l-ai poluat cu concern-uri de prezentare. Nu i se extinde prea mult responsabilitatea daca trebuie sa intrebe utilizatorul?